extensions/sql-migration/src/dashboard/sqlServerDashboard.ts

```diff
@@ -46,6 +49,16 @@ export class DashboardWidget {
 		});
 	}
 
+	private async getCurrentMigrations(): Promise<MigrationContext[]> {
+		const connectionId = (await azdata.connection.getCurrentConnection()).connectionId;
+		return this._migrationStatusMap.get(connectionId)!;
+	}
+
+	private async setCurrentMigrations(migrations: MigrationContext[]): Promise<void> {
+		const connectionId = (await azdata.connection.getCurrentConnection()).connectionId;
```

Comment:
I would expect that the connection ID is never going to change once a dashboard is open - is there any reason you can't just store it when you create the dashboard?

(although I'm not sure if we recreate the dashboard widgets when switching between databases and servers on a dashboard using the breadcrumb navigator or something. So maybe we do have to do this each time if we don't recreate the widgets)

Reply:
Actually, we don't at this point in time. Therefore any class variable is shared between different dashboard widgets.  This made implement this solution.